xcsv.cc

```diff
@@ -253,9 +258,15 @@ XcsvStyle::xcsv_ofield_add(XcsvStyle* style, const QString& qkey, const QString&
   QByteArray val = qval.toUtf8();
   QByteArray pfc = qpfc.toUtf8();
 
+#if 0
   struct xt_mapping* xm = Perfect_Hash::in_word_set(key.constData(), strlen(key.constData()));
-
   field_map fmp(key, val, pfc, xm ? xm->xt_token : -1, options);
+#else
+  if (!xcsv_tokens.contains(key)) {
+    warning("Missing key: %s\n", CSTR(qkey));
+  }
+  field_map fmp(key, val, pfc, xcsv_tokens.value(qkey));
```

Comment:
perhaps with 
xcsv_tokens.value(qkey, XT_unused)
it is easier to understand what is returned if the key isn't found.  What is the default constructed value of an enum?  Qt says `But for primitive types like int and double, as well as for pointer types, the C++ language doesn't specify any initialization; in those cases, Qt's containers automatically initialize the value to 0.
`
and it looks like qhash.h achives this by using zero initialization.  So I think your existing code works and isn't relying on undefined behavior, but adding being explicit seems easier to understand.

Reply:
Agreed. This code has had a hard life.